lib/modules/apostrophe-migrations/lib/api.js

```diff
@@ -81,20 +87,66 @@ module.exports = function(self, options) {
       // affect the remainder of the query.
       //
       // https://groups.google.com/forum/#!topic/mongodb-user/AFC1ia7MHzk
-      cursor.sort({ _id: 1 });
-      return broadband(cursor, limit, function(doc, callback) {
-        if (iterator.length === 1) {
-          return Promise.try(function() {
-            return iterator(doc);
-          }).then(function() {
-            return callback(null);
-          }).catch(function(err) {
+
+      var progressDisplay = process.stdout.isTTY && (self.options.progressDisplay !== false);
+      var progress;
+
+      return async.series([ count, execute ], callback);
+
+      function count(callback) {
+        if (!progressDisplay) {
+          return callback(null);
+        }
+        if (progressDisplay) { 
+          console.error('Determining total...');
```

Comment:
console.error because the progress display goes there too, leaving console.log clean.